src/Commands/Provisioning/Site/AddDataRowsToSiteTemplate.cs

```diff
@@ -79,6 +83,11 @@ protected override void ExecuteCmdlet()
                 throw new ApplicationException("List does not exist in the template file!");
             }
 
+            if (string.IsNullOrEmpty(KeyColumn))
+            {
+                listInstance.DataRows.KeyColumn = KeyColumn;
```

Comment:
Can you elaborate on what the KeyColumn does?

Reply:
Hi @KoenZomers ,

KeyColumn allows us to leverage the UpdateListItem() under [pnp/pnpframework](https://github.com/pnp/pnpframework/blob/68de62097883124357c3bcdfb2b17e789f1827d7/src/lib/PnP.Framework/Provisioning/ObjectHandlers/Utilities/ListItemUtilities.cs#L264) repository.

It allows us to basically overwrite already existing sharepoint list items when performing "Add-PnPDataRowsToSiteTemplate" operation. 

KeyColumn parameter helps us to define the primary-key of the sharepoint list in order to identify unique list items.

Under pnp/pnpframework repository, KeyColumn is parsed under the [ObjectListInstanceDataRows.cs](https://github.com/pnp/pnpframework/blob/dev/src/lib/PnP.Framework/Provisioning/ObjectHandlers/ObjectListInstanceDataRows.cs#L51) class.

Reply:
Thanks for clarifying